src/components/SearchInput/SearchInput.jsx

```diff
@@ -1,7 +1,62 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+
+import styles from './SearchInput.module.scss';
+import SearchIcon from '../../assets/images/search-icon.png';
+import SearchLocationIcon from '../../assets/images/search-location-icon.png';
+import { NUMBER_OF_RESULTS } from '../../constants/forecaApi';
+import { dataService } from '../../services/dataService';
 
 function SearchInput() {
-  return <div>Search input</div>;
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+
+  useEffect(() => {
+    if (searchQuery) {
+      searchCities();
+    }
+  }, [searchQuery]);
```

Comment:
Also as an improvement - think how can you optimize your code. Right now on every key stroke you are sending a request. Could you **debounce** it?